### PR TITLE
Use argocd to manage utilities for e2e-cluster

### PR DESCRIPTION
**Pull request by andrleite**
#### Summary
Use argocd to manage utilities for e2e-cluster

#### Ticket Link
 Fixes https://mattermost.atlassian.net/browse/CLD-7783

#### Release Note

```release-note
None
```

### CODE REVIEW

File: e2e/tests/shared/shared.go [hunk 1]
```diff
@@ -100,6 +100,57 @@ func SetupTestWithDefaults(testName string) (*Test, error) {
...
+			AMI:                config.KopsAMI,
+			VPC:                config.VPC,
+			Provisioner:        config.Provisioner,
+			ArgocdClusterRegister: map[string]string{
+				"cluster-type": "customer",
+			},
+			DesiredUtilityVersions: map[string]*model.HelmUtilityVersion{
+				"nginx": {
+					Chart: "unmanaged",
+				},
```
Comment (gabrieljackson) [inline e2e/tests/shared/shared.go:116]: We should use the constants for all these. Example:
```suggestion
				model.NginxCanonicalName: {
					Chart: model.UnmanagedUtilityVersion,
				},
```

File: e2e/tests/shared/shared.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ type Test struct {
 	Cleanup           bool
 }
 
-func SetupTestWithDefaults(testName string) (*Test, error) {
+func SetupTestWithDefaults(testName, utiliyManaged string) (*Test, error) {
```
Comment (gabrieljackson) [inline e2e/tests/shared/shared.go:76]: I think passing in `utiliyManaged` as a bool makes a bit more sense than as a string.

```suggestion
func SetupTestWithDefaults(testName string, argoManagedUtilities bool) (*Test, error) {
```

File: e2e/tests/shared/shared.go [hunk 3]
```diff
@@ -100,6 +100,57 @@ func SetupTestWithDefaults(testName string) (*Test, error) {
 		Provisioner:        config.Provisioner,
 	}
 
+	if utiliyManaged == "argocd" {
+		createClusterReq = &model.CreateClusterRequest{
+			AllowInstallations: true,
+			Annotations:        testAnnotations(testID),
+			AMI:                config.KopsAMI,
+			VPC:                config.VPC,
+			Provisioner:        config.Provisioner,
```
Comment (gabrieljackson) [inline e2e/tests/shared/shared.go:109]: Instead of recreating the request we can just alter the existing one to change or add the values we want.

**Review by gabrieljackson — CHANGES_REQUESTED** [review]
I think we can make a few improvements which I outlined below

**Review by nickmisasi — COMMENTED** [review]
Nothing to add from Gabe's suggestions, just commenting so you can re-request review when you've addressed them!

**Review by gabrieljackson — APPROVED** [review]
LGTM. Thanks for addressing the feedback.

**Pull request merged**